### PR TITLE
Impl Login TC: 05,06,08

### PR DESCRIPTION
**Pull request by sattdn-23-01-03-NguyenNhatNam**
Please review me

### CODE REVIEW

File: src/main/java/page/ChangePassword.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+package page;
+
+import common.Constant;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+public class ChangePassword {
+    private By txtCurentPw = By.xpath("//input[@id='currentPassword']");
```
Comment (cheanhminh98) [inline src/main/java/page/ChangePassword.java]: please update the variable for full meaning

File: src/main/java/page/ChangePassword.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    private By txtNewPw = By.xpath("//input[@id='newPassword']");
+    private By txtConfirmPw = By.xpath("//input[@id='confirmPassword']");
+    private By btnChangePw = By.xpath("//form[@id='ChangePW']//input[@type='submit']");
+    private By msgSuccess = By.xpath("//p[@class='message success']");
+
+    public WebElement getCurrentPwField() {
```
Comment (cheanhminh98) [inline src/main/java/page/ChangePassword.java]: please update the method for full meaning

File: src/main/java/common/Constant.java [hunk 3]
```diff
@@ -5,4 +5,9 @@
 public class Constant {
     public static final String RAILWAY_URL = "http://www.railwayb2.somee.com/Page/HomePage.cshtml";
     public static WebDriver WEBDRIVER ;
+    public static final String Username = "nhatnam@gmai.com";
+    public static final String Password = "12345678";
+
+
```
Comment (cheanhminh98) [inline src/main/java/common/Constant.java:12]: remove redundant lines

File: src/main/java/page/GeneralPage.java [hunk 4]
```diff
@@ -2,26 +2,71 @@
...
 public class GeneralPage {
-    private WebDriver driver;
-    private By tabLogin = By.xpath("//a[contains(@href,'Login')]");
-    private By tabRegister = By.xpath("//a[contains(@href,'Register')]");
+    private String Login = "Login";
+    private String Register = "Register";
+    private String BookTicket = "BookTicketPage";
+    private String Logout = "Logout";
+    private String MyTicket = "ManageTicket";
+    private String ChangePassWord = "ChangePassword";
```
Comment (cheanhminh98) [inline src/main/java/page/GeneralPage.java:18]: please use them (Login/Register/...) in the enum or constant

File: src/main/java/page/GeneralPage.java [hunk 5]
```diff
@@ -2,26 +2,71 @@
...
+    }
+
+    public void clickOnTabLogout() {
+        this.getTabLogout().click();
+    }
+
+    public void clickOnTabChangPassword() {
+        this.getChangePassword().click();
+    }
+    public void clickOnTagMyTicket(){
```
Comment (cheanhminh98) [inline src/main/java/page/GeneralPage.java]: format code

File: src/test/resources/TestRun.xml [hunk 6]
```diff
@@ -3,9 +3,9 @@
 <suite name="Practice">
     <test name="Test1">
         <classes>
-            <class name="railway.RegisterTest">
+            <class name="railway.ChangePwTest">
```
Comment (cheanhminh98) [inline src/test/resources/TestRun.xml:6]: Why do you always edit this one?

File: src/main/java/page/RegisterPage.java [hunk 7]
```diff
@@ -2,67 +2,107 @@
 
 import common.Constant;
 import org.openqa.selenium.*;
+import utils.Utilities;
 
 public class RegisterPage {
 
-    public GeneralPage dashboardPage;
+    public Utilities utilities = new Utilities();
```
Comment (cheanhminh98) [inline src/main/java/page/RegisterPage.java:9]: why do you call the Utilities() here?

File: src/test/resources/TestRun.xml (hunk 6 above)
Comment (sattdn-23-01-03-NguyenNhatNam) [reply src/test/resources/TestRun.xml:6]: Because I want to run exactly test case that I choose

File: src/main/java/page/GeneralPage.java (hunk 4 above)
Comment (sattdn-23-01-03-NguyenNhatNam) [reply src/main/java/page/GeneralPage.java:18]: I will create a enum class to contain it

File: src/main/java/common/Enums.java [hunk 8]
```diff
@@ -0,0 +1,2 @@
+package common;public enum Enu {
```
Comment (cheanhminh98) [inline src/main/java/common/Enums.java:1]: enter a line

File: src/main/java/page/ChangePasswordPage.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
+package page;
+
+import common.Constant;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+public class ChangePasswordPage {
+    private By txtCurentPassword = By.xpath("//input[@id='currentPassword']");
```
Comment (cheanhminh98) [inline src/main/java/page/ChangePasswordPage.java:8]: we should use the dynamic locator

File: src/test/java/railway/ChangePasswordTest.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+//        homePage.clickOnTabChangPassword();
+//        String newPw = "namdeptrai"+ utilities.randomNumber(1,999)+"a@";
+//        System.out.println("newPW:"+newPw);
+//        String currentPw = utilities.changeCurrentPw(Constant.Password,newPw);
+//        String confirmPw = newPw;
+//        changePassword.changePw(currentPw,newPw,confirmPw);
+//
+//        Assert.assertEquals(changePassword.showSuccessMsg(), "Your password has been updated");
+//
+//    }
```
Comment (cheanhminh98) [inline src/test/java/railway/ChangePasswordTest.java:28]: please remove redundant codes

File: src/test/java/railway/LoginTest.java [hunk 11]
```diff
@@ -1,53 +1,98 @@
...
 import page.LoginPage;
 
 public class LoginTest extends BaseTest {
     public HomePage homePage = new HomePage();
     public LoginPage logInPage = new LoginPage();
+    public GeneralPage generalPage = new GeneralPage();
 
     @Test
     public void TC01() {
         System.out.println("User can login Railway with valid username and password ");
```
Comment (cheanhminh98) [inline src/test/java/railway/LoginTest.java:17]: please use log4j instead

File: src/test/java/railway/RegisterTest.java [hunk 12]
```diff
@@ -12,23 +12,58 @@ public class RegisterTest extends BaseTest {
     public HomePage homePage = new HomePage();
 
     @Test
-    public void TC() throws InterruptedException {
+    public void TC00() throws InterruptedException {
```
Comment (cheanhminh98) [inline src/test/java/railway/RegisterTest.java:15]: why is this TC00?
and do not use throws exception here

File: src/test/java/railway/RegisterTest.java [hunk 13]
```diff
@@ -12,23 +12,58 @@ public class RegisterTest extends BaseTest {
...
+        String actualMsg = registerPage.showSuccessMsg();
+        Assert.assertEquals(actualMsg, "You're here", "Register failure, Account already exists");
 
         homePage.clickOnTabLogin();
         loginPage.login(email, password);
 
     }
 
+    @Test
+    public void TC07() throws InterruptedException {
```
Comment (cheanhminh98) [inline src/test/java/railway/RegisterTest.java:34]: same as above

File: src/test/java/railway/RegisterTest.java [hunk 14]
```diff
@@ -12,23 +12,58 @@ public class RegisterTest extends BaseTest {
...
+        homePage.open();
+        homePage.clickOnTabRegister();
+        registerPage.register("nam3@gmail.com", "123456789", "123456789", "123123123");
+        String actualMsg = registerPage.showSuccessMsg();
+        Assert.assertEquals(actualMsg, "Thank you for registering your account", "Register failure, Account already exists");
+    }
+
+
+    @Test
+    public void TC10() throws InterruptedException {
```
Comment (cheanhminh98) [inline src/test/java/railway/RegisterTest.java:44]: same as above

File: src/test/java/railway/RegisterTest.java [hunk 15]
```diff
@@ -12,23 +12,58 @@ public class RegisterTest extends BaseTest {
...
+        String confirmPassword = "namratdeptrai";
+        registerPage.register("nam6@gmail.com", password, confirmPassword, "123123123");
+
+        Assert.assertEquals(registerPage.showErrorRegisterMsg(), "There're errors in the form. Please correct the errors and try again.", "The error message doesn't display properly");
+
+        Assert.assertEquals(registerPage.showErrorConfirmPwMsg(), "The two passwords do not match");
+    }
+
+    @Test
+    public void TC11() throws InterruptedException {
```
Comment (cheanhminh98) [inline src/test/java/railway/RegisterTest.java:57]: same as above

File: src/test/resources/TestRun.xml (hunk 6 above)
Comment (cheanhminh98) [reply src/test/resources/TestRun.xml:6]: So when you want to run an any test case you will edit it?

File: src/main/java/page/GeneralPage.java [hunk 16]
```diff
@@ -1,27 +1,73 @@
...
+
+    public void clickOnTabLogout() {
+        this.getTabLogout().click();
+    }
+
+    public void clickOnTabChangPassword() {
+        this.getChangePassword().click();
+    }
+
+    public void clickOnTagMyTicket() {
```
Comment (cheanhminh98) [inline src/main/java/page/GeneralPage.java:69]: please edit Tag to Tab

**Pull request merged**